### PR TITLE
add upgrade_bcd.md [BCD-281]

### PR DESCRIPTION
**Pull request by JeremJR**

### CODE REVIEW

File: md/upgrade_bcd.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Migrate from an earlier version of BCD
+
+## Overview
+
+This page explains how to upgrade your version of BCD.
+
+Each versions follow the MAJOR.MINOR.PATCH format.
```
Comment (alarreine) [inline md/upgrade_bcd.md]: Each + singular or
Each of the ..

Comment (alarreine) [inline md/upgrade_bcd.md]: Each + singular or
Each of the ..

File: md/upgrade_bcd.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+On each steps we will specify if there are differences for a major upgrade (1.0.x to 2.0.y) or a patch (2.0.x to 2.0.y).
```
Comment (alarreine) [inline md/upgrade_bcd.md]: Each + singular or
Each of the ..

File: md/upgrade_bcd.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+- Terraform metadata
+- bcd controller
+- Vagrant
+- Living Application repositories
+
+## How it works
+
+### Scenarios
+
+First you may want to keep your existing scenarios. It will allow you to recreate the same kind of stack or manipulate an existing one (deploy, undeploy, destroy).
```
Comment (alarreine) [inline md/upgrade_bcd.md]: First, ....

Maybe?

File: md/upgrade_bcd.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+If you migrate from a 1.0.x towards a 2.0.y you will first need to copy the files
+```
+cp bonita-continuous-delivery_1.0.5/scenarios/*.yml bonita-continuous-delivery_2.0.0/scenarios/
+```
+Then you will have to check `Breaking changes` section into the [release notes](release_notes.md). Indeed you may want to override `bonita_http_api` default value that have changed. Or for deployments that do not use `aws` for `bcd_provider` you will need to replace `on-premises` by `static_inventory`.
```
Comment (alarreine) [inline md/upgrade_bcd.md]: ...`bonita_http_api` default value that __has__

File: md/upgrade_bcd.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+If you migrate from a 2.0.x towards a 2.0.y you will only need to copy the files.
+
+```
+cp bonita-continuous-delivery_2.0.0/scenarios/*.yml bonita-continuous-delivery_2.0.2/scenarios/
+```
+
+### Terraform
+
+"Terraform must store state about your managed infrastructure and configuration. This state is used by Terraform to map real world resources to your configuration, keep track of metadata, and to improve performance for large infrastructures." [https://www.terraform.io/docs/state/](https://www.terraform.io/docs/state/).
```
Comment (alarreine) [inline md/upgrade_bcd.md]: * The first " has not been closed
* Terraform must store __the__ state

File: md/upgrade_bcd.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+So if you want to be able to manage an existing stack (ie a stack already deployed from a previous version of BCD) you will need to copy paste this state.
```
Comment (alarreine) [inline md/upgrade_bcd.md]: * i.e.

File: md/upgrade_bcd.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+cp bonita-continuous-delivery_1.0.5/terraform/your_stack_name.tfstate bonita-continuous-delivery_2.0.0/terraform/your_stack_name/your_stack_name.tfstate
+```
+If you migrate from a 2.0.x towards a 2.0.y you will only need to copy the directory corresponding to your stack.
+```
+cp -r bonita-continuous-delivery_2.0.0/terraform/your_stack_name bonita-continuous-delivery_2.0.2/terraform/
+```
+
+### BCD controller
+
+Like described into the "Installation guide" from the [Getting started](getting_started.md) you will need to load the last version of `bcd-controller_<version>.tar.zip Docker image.
```
Comment (alarreine) [inline md/upgrade_bcd.md]: As is described..

**Review by alarreine — COMMENTED** [review]
I propose the example of  [Ansible](https://docs.ansible.com/ansible/2.5/porting_guides/porting_guide_2.0.html). We can find in each doc the porting guide

**Comment by JeremJR** [comment]
thanks @alarreine I've applied the first changes, let see with @tnguyen1 if he is ok with the rework like Ansible porting guides

**Comment by JeremJR** [comment]
I've try to reorganize the page as discussed.

**Review by alarreine — APPROVED** [review]

**Pull request merged**